### PR TITLE
feat(crd): Mebius Machine CRD design

### PR DESCRIPTION
**Pull request by LavenderQAQ**
#### What type of PR is this?
/kind design

#### What this PR does / why we need it:
Defines the basic structure of machine crd.

### CODE REVIEW

File: pkg/gateway/gateway.go [hunk 1]
```diff
@@ -52,8 +53,11 @@ func (g *gateway) Run() error {
 			return md
 		}),
 	)
-	dialOption := []grpc.DialOption{grpc.WithInsecure()}
-	protos.RegisterServerHandlerFromEndpoint(ctx, mux, g.opts.Gateway().Endpoints, dialOption)
+	dialOption := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
```
Comment (LavenderQAQ) [inline pkg/gateway/gateway.go:56]: grpc.WithInsecure is deprecated: use WithTransportCredentials and insecure.NewCredentials() instead. Will be supported throughout 1.x.

File: pkg/context/context.go [hunk 2]
```diff
@@ -76,8 +76,8 @@ func (m *mcontext) Err() error {
 
 func (m *mcontext) GO(exec func()) {
 	wg := m.Value(_waitGroupKey{}).(*sync.WaitGroup)
+	wg.Add(1)
 	go func() {
-		wg.Add(1)
```
Comment (LavenderQAQ) [inline pkg/context/context.go:80]: Data race occurs during unit test

File: pkg/utils/function/link.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ func (f *FunctionLink) Do() {
 	}
 	if f.fle != nil {
 		for _, fc := range *f.fle {
-			fc()
+			_ = fc()
```
Comment (LavenderQAQ) [inline pkg/utils/function/link.go:48]: fix lint check of ci workflow

**Review by LTitan — APPROVED** [review]
LGTM

**Pull request merged**